### PR TITLE
Added Travis CI to project

### LINKED ISSUE #102: Add Travis CI
In order to avoid having to run our deploy scripts each time we make a new change to the `master` branch, we should setup some Continuous Integration to do this job for us. 

[Travis CI](https://travis-ci.org/) can accomplish this for us with the added benefit of integrating with GitHub, indicating whether any given commit causes the project build to fail (bad exit code from any deploy script) and preventing the merging of failing branches into `master`.

### PR DESCRIPTION
**Pull request by Davidster**
resolves #102 

## Summary

I've added Travis CI to the project which, as explained in #102, builds the project every time a commit is added to any branch in the repo and deploys (transfers war file) the project after building for all commits made to `master`. The build gets triggered every time you use `git push` with some new changes.

In order to have a distinction between building and deploying the project ([Travis is made to work well with such a setup](https://docs.travis-ci.com/user/customizing-the-build/#The-Build-Lifecycle)), I added some new scripts (see `scripts/`) which do these tasks individually and changed `deploy.sh` so that it just runs said scripts. ~**The reason I made this PR "help wanted" is because I wanted to know if there is an easy way to eliminate the duplicated code of the first part of `deployApp.sh` and `buildApp.sh` where the command line arguments are parsed. Any help on that front would be appreciated.**~ - Resolved by this [commit](/stumash/CoursePlanner/pull/103/commits/716149f2ef3b027bfb18d79b87545508333abcae)

I left `deploy.sh` in the root of the project in order to avoid any confusion - I didn't see the need to change the development process that were currently used to: 1. Change some code, 2. Run `deploy.sh`. 

Note that if you want to run `buildApp.sh` or `deployApp.sh` individually, you must do so from the root of the project like so: `sh scripts/buildApp.sh` since the scripts still assume they're in the root directory. 

## The `deploy_rsa.enc` file
Just as we all need a private rsa key to gain access to the machine hosting our project, so does Travis.org.  We give it the key by literally putting the key in a file in our Github repository.  However, since this is obviously insecure on a publicly-hosted repo, we don't actually put the bare rsa-key file in repo -- we put an encrypted version of it!

In order to encrypt the rsa-key file such that Travis.org can decrypt it and use it to access the server, we use a`travis cli` that can be downloaded for free online. The cli first asks us to login (to verify that we are truly contributors to the repo) and then lets us encrypt whatever we want, sending Travis.org the info it will need to decrypt whatever we encrypt.

So, the `deploy_rsa.enc` file is an encrypted file, which, when travis decrypts it, contains the ssh key to our server.  All the commands in the `before_deploy` section of the `.travis.yml` file simply tell travis to decrypt the file and then add the ssh-key to its ssh-agent.

I read about how to set that up in these two articles:
https://oncletom.io/2016/travis-ssh-deploy/
https://docs.travis-ci.com/user/encrypting-files/

## Test
In this case, the test is the merge. We should find that `/opt/tomcat/webapps/Root.war` on the server gets updated a couple of minutes after this PR is merged. Also, all Travis output can be viewed [here](https://travis-ci.org/stumash/CoursePlanner/).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+before_script:
+- sudo apt-get install nodejs
```
Comment (PeterGhimself) [inline .travis.yml]: I guess there isn't any way around this, but isn't it generally bad practice to have `sudo`s in scripts? @stumash

File: scripts/buildApp.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            exit 1
+        fi
+    fi
+else
+    echo "deploy.sh takes one required argument: the name of the dev or \"prod\""
+    echo "deploy.sh takes an optional second argument: \"--verbose\" to print npm and maven output"
+    exit 1
+fi
+
+echo ""
```
[the remark this answers is not in the document]
Comment (PeterGhimself) [reply scripts/buildApp.sh]: oh it makes a new line!

File: deploy.sh [hunk 3]
```diff
@@ -1,100 +1,10 @@
...
-    finalWarName="ROOT.war"
-else
-    finalWarName="courseplanner${devname}.war"
-fi
-echo "copying .war file to server..."
-if eval $deployToServerCommand$finalWarName $verbose # run the deployment command and react to exit code
+buildAndDeploy="./scripts/buildApp.sh $* && ./scripts/deployApp.sh $*"
+if eval $buildAndDeploy
 then
-    echo "copying ${finalWarName} to server successful at: $(date)"
```
[the remark this answers is not in the document]
Comment (PeterGhimself) [reply deploy.sh]: oh you just moved it to `deployApp.sh`

File: .travis.yml (hunk 1 above)
Comment (PeterGhimself) [reply .travis.yml]: confirmed

**Comment by stumash** [comment]
how do we feel about renaming `deploy.sh`.  I just think it's odd now that it does building and deploying...
how about `buildAndDeploy.sh` or something?

**Comment by Davidster** [comment]
If we are to change to name, I vote for `buildAndDeploy.sh`

**Pull request merged**